### PR TITLE
Batch endpoint for Dataset Information Service

### PR DESCRIPTION
**Pull request by mephenor**

### CODE REVIEW

File: 54-peacock-spider/technical_specification.md [hunk 1]
```diff
@@ -38,17 +43,71 @@ hiding unnecessary details for the requester.
...
+File information can exist in one of three states:
+
+1) The requested file is not yet registered
+2) The requested file is registered
+3) The requested file is deleted
+
+If the reason why data is not available is of no interest and if the all dataset registered states are treated the same, the following responses will be returned:
+
+1) The file information endpoint either returns a payload or 404
+2) The dataset information returns either a payload or 404. Within the payload, the fields of each file information object either are populated with the actual data or a None/null placeholder value. The API consumer can then check just one of the non accession fields to evaluate if a file information object is populated.
```
Comment (mephenor) [inline 54-peacock-spider/technical_specification.md:109]:
```suggestion
1) The file information endpoint either returns a payload or 404.
2) The dataset information endpoint returns either a payload or 404. 
Within the payload, the fields of each file information object either are populated with the actual data or a None/null placeholder value.
The API consumer can then check just one of the non accession fields to evaluate if a file information object is populated. 
```

File: 54-peacock-spider/technical_specification.md [hunk 2]
```diff
@@ -33,22 +38,81 @@ This should return a payload including the unencrypted file size in bytes and SH
...
+4) The requested dataset is registered and all file information is available
+5) The requested dataset is deleted
+
+File information can exist in one of three states:
+
+1) The requested file is not yet registered
+2) The requested file is registered
+3) The requested file is deleted
+
+If the reason why data is not available is of no interest and if the all dataset registered states are treated the same, the following responses will be returned:
```
Comment (mephenor) [inline 54-peacock-spider/technical_specification.md:110]:
```suggestion
If the reason why data is not available is of no interest and if all dataset registered states are treated the same, the following responses will be returned:
```

File: 54-peacock-spider/technical_specification.md [hunk 3]
```diff
@@ -33,22 +38,81 @@ This should return a payload including the unencrypted file size in bytes and SH
...
+File information can exist in one of three states:
+
+1) The requested file is not yet registered
+2) The requested file is registered
+3) The requested file is deleted
+
+If the reason why data is not available is of no interest and if all dataset registered states are treated the same, the following responses will be returned:
+
+1) The file information endpoint either returns a payload or 404
+2) The dataset information returns either a payload or 404.
```
Comment (mephenor) [inline 54-peacock-spider/technical_specification.md:113]:
```suggestion
2) The dataset information endpoint returns either a payload or 404.
```

**Review by lkuchenb — APPROVED** [review]
Thank you, LGTM!

**Pull request merged**